### PR TITLE
fix(DATAGO-123915): removing generated files from project artifact count

### PR DESCRIPTION
**Pull request by lgh-solace**
### What is the purpose of this change?

    Fixing project artifact count to not include files generated for indexing

### How was this change implemented?

This pull request introduces a new utility function to consistently identify internal/generated artifact files and updates the artifact counting logic to use this function, ensuring that only user-uploaded files are counted as artifacts. It also adds comprehensive unit tests for both the new helper and the updated counting logic.

**Artifact file classification and counting improvements:**

* Added the `is_internal_artifact` function to `artifact_helpers.py` to determine if a filename represents an internal/generated file (metadata, converted text, or BM25 index files). This centralizes and standardizes the logic for identifying such files.
* Updated the artifact counting logic in `get_artifact_counts_batch` to use `is_internal_artifact`, ensuring that only user-uploaded files are counted as artifacts, and clarified the docstring accordingly. [[1]](diffhunk://#diff-5ae8053066a35d07d6335108bb6d978ca59bf3c0b348a25e814802a083bc02e7L987-R1007) [[2]](diffhunk://#diff-5ae8053066a35d07d6335108bb6d978ca59bf3c0b348a25e814802a083bc02e7L1000-R1021)
* Refactored `_is_original_artifact` in `project_service.py` to delegate to `is_internal_artifact`, reducing duplication and improving maintainability.
* Updated imports to include the new helper function where needed. [[1]](diffhunk://#diff-68a8e7a0f2adb218d6d54d4f8831802929eb24da553a75d143c915ffb8d9eb56L14-R14) [[2]](diffhunk://#diff-68f9376fb7099e63f939f885afb73e27452d3393e2beddf29ec65e00f85ffb84R17) [[3]](diffhunk://#diff-68f9376fb7099e63f939f885afb73e27452d3393e2beddf29ec65e00f85ffb84R29)

**Testing enhancements:**

* Added a new test class for `is_internal_artifact` covering various edge cases and typical filenames, and a test class for `get_artifact_counts_batch` to ensure correct artifact counting behavior.

### How was this change tested?

- [x] Manual testing: [describe scenarios]
- [x] Unit tests: [new/modified tests]
- [ ] Integration tests: [if applicable]
- [ ] Known limitations: [what wasn't tested]

### Is there anything the reviewers should focus on/be aware of?

    Special attention areas, potential risks, or open questions

### CODE REVIEW

File: src/solace_agent_mesh/agent/utils/artifact_helpers.py [hunk 1]
```diff
@@ -997,8 +1017,8 @@ async def get_artifact_counts_batch(
                 keys = await list_keys_method(
                     app_name=app_name, user_id=user_id, session_id=session_id
                 )
-                # Count only non-metadata files
-                count = sum(1 for key in keys if not key.endswith(METADATA_SUFFIX))
```
Comment (lgh-solace) [inline src/solace_agent_mesh/agent/utils/artifact_helpers.py:1001]: This count was only excluding the metadata files and not the converted files so the count returned to the UI was incorrect.

File: src/solace_agent_mesh/agent/utils/artifact_helpers.py [hunk 2]
```diff
@@ -33,10 +33,30 @@
 log = logging.getLogger(__name__)
 
 METADATA_SUFFIX = ".metadata.json"
+CONVERTED_TEXT_SUFFIX = ".converted.txt"
+BM25_INDEX_FILENAME = "project_bm25_index.zip"
 DEFAULT_SCHEMA_MAX_KEYS = 20
 DEFAULT_SCHEMA_INFERENCE_DEPTH = 4
 
 
+def is_internal_artifact(filename: str) -> bool:
```
Comment (lgh-solace) [inline src/solace_agent_mesh/agent/utils/artifact_helpers.py:41]: Created a common function so it can be shared in this file and in the project service.

File: src/solace_agent_mesh/agent/utils/artifact_helpers.py [hunk 3]
```diff
@@ -997,8 +1017,8 @@ async def get_artifact_counts_batch(
                 keys = await list_keys_method(
                     app_name=app_name, user_id=user_id, session_id=session_id
                 )
-                # Count only non-metadata files
-                count = sum(1 for key in keys if not key.endswith(METADATA_SUFFIX))
+                # Count only user-uploaded files (exclude metadata, converted text, and index files)
+                count = sum(1 for key in keys if not is_internal_artifact(key))
```
[the remark this answers is not in the document]
Comment (lgh-solace) [reply src/solace_agent_mesh/agent/utils/artifact_helpers.py:1022]: I left this comment for consideration, Rudraneel. It seems like these files are not "yet" created, but I'm not sure if the guard should be added now. Thoughts?

File: src/solace_agent_mesh/agent/utils/artifact_helpers.py (hunk 2 above)
Comment (rudraneel-chakraborty) [inline src/solace_agent_mesh/agent/utils/artifact_helpers.py:41]: This function can be simplified to 
```
def is_internal_artifact(filename: str) -> bool:
      """..."""
      return (
          filename.endswith(METADATA_SUFFIX) or
          filename.endswith(CONVERTED_TEXT_SUFFIX) or
          filename == BM25_INDEX_FILENAME
      )

```

File: src/solace_agent_mesh/agent/utils/artifact_helpers.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rudraneel-chakraborty) [reply src/solace_agent_mesh/agent/utils/artifact_helpers.py:1022]: I think this is a valid point, although this is not relevant now but no harm in fixing this. 
alternatively, you could strip of the `user:` prefix in `is_internal_artifact` method

File: src/solace_agent_mesh/agent/utils/artifact_helpers.py (hunk 2 above)
Comment (rudraneel-chakraborty) [reply src/solace_agent_mesh/agent/utils/artifact_helpers.py:41]: I think there are a couple of places where this utility function can be used, e.g. https://github.com/SolaceLabs/solace-agent-mesh/blob/4411f548fe6dda55d270d2362d54b7789971a731/src/solace_agent_mesh/gateway/http_sse/utils/artifact_copy_utils.py#L184
